modules/core/04-channel/types/upgrade.go

```diff
@@ -77,12 +77,48 @@ func (ut Timeout) IsValid() bool {
 	return !ut.Height.IsZero() || ut.Timestamp != 0
 }
 
+// UpgradeError defines an error that occurs during an upgrade.
+type UpgradeError struct {
+	// underlyingError is the underlying error that caused the upgrade to fail.
+	// this error should not be written to state.
+	underlyingError error
+	// upgradeSequence is the sequence number of the upgrade that failed.
+	upgradeSequence uint64
+}
+
+func (u *UpgradeError) Error() string {
```

Comment:
I think this needs to match the return value of `NewUpgradeError`? When testing I got this error:
> keeper/upgrade.go:172:10: cannot use types.NewUpgradeError(channel.UpgradeSequence, errorsmod.Wrapf(types.ErrIncompatibleCounterpartyUpgrade, "expected upgrade sequence (%d) to match counterparty upgrade sequence (%d)", channel.UpgradeSequence, counterpartyChannel.UpgradeSequence)) (value of type "[github.com/cosmos/ibc-go/v7/modules/core/04-channel/types](http://github.com/cosmos/ibc-go/v7/modules/core/04-channel/types)".UpgradeError) as error value in return statement: "[github.com/cosmos/ibc-go/v7/modules/core/04-channel/types](http://github.com/cosmos/ibc-go/v7/modules/core/04-channel/types)".UpgradeError does not implement error (method Error has pointer receiver)

Until I removed the pointer receiver in this function and the error receipt one

Reply:
ended up updating to remove method receivers, tests seem to be passing now, let me know if there are any other issues